avocado/utils/ssh.py

```diff
@@ -248,3 +249,23 @@ def copy_files(self, source, destination, recursive=False):
             return result.exit_status == 0
         except process.CmdError as exc:
             raise NWException("failed to copy file {}".format(exc))
+
+    def remove_control_path_reconnect(self):
+        """
+        Establishes connection to the remote host if failed try to
+        close the session.if it failed remove control path file and
+        reconnect it.
+
+        :returns: True if success or else raise exception
+        :rtype: bool
+        """
+        if not self.connect():
+            if not self.quit():
+                full_path = os.path.expanduser('~/.ssh')
+                file_list = glob.glob("{}/avocado-master-root*".format(full_path))
+                for path in file_list:
+                    if self.host in path:
+                        os.remove(path)
```

Comment:
Same comment as before: you are trying to remove others' control files.

Imagine that we have multiple tests, with multiple host connections, and because of one connection you are removing all control files.

Reply:
@beraldoleal now i am removing one control path file,based on ip address.

Reply:
Ohh I see now, sorry for not noticing before.

But still, this is a little strange to me. Let me take a proper look here, and see what we can do.

Reply:
Ok, @smruti77, I did a closer look here to see what we can do. 

Your PR is assuming here that all connections are made to remote users 'root'. When, in fact, a different user can be used. Also, you are removing all files that have the same host, regardless of the port. IMO, this is a little over needed.

The reconnection portion is a specific use case, and different users can have different needs.

I have created a new PR adding one new method and one new property.  Users could run the cleanup of control files if needed. Please take a look at https://github.com/avocado-framework/avocado/pull/4350 and let me know if this will work for you.